framework/src/base/MooseInit.C

```diff
@@ -21,6 +21,22 @@
 #include <omp.h>
 #endif
 
+#include <unistd.h>
+#include <signal.h>
+
+void
+my_sig_handler(int signum)
+{
+  Moose::autosave_flag = signum;
+  return;
```

Comment:
Not a problem, but why?

Reply:
In the future, we wanted to be able to differentiate between signals -- if we just make the global variable the same as the signal, then we can granularly act based on the value of the global variable anywhere else.

Reply:
The return statement isn't needed. 

Also not requiring you to remove it.